src/ngx_http_lua_directive.c

```diff
@@ -936,13 +954,17 @@ ngx_http_lua_header_filter_by_lua(ngx_conf_t *cf, ngx_command_t *cmd,
         /* Don't eval nginx variables for inline lua code */
         llcf->header_filter_src.value = value[1];
 
-        p = ngx_palloc(cf->pool, NGX_HTTP_LUA_INLINE_KEY_LEN + 1);
+        p = ngx_palloc(cf->pool,
+                       sizeof("header_filter_by_lua") +
+                       NGX_HTTP_LUA_INLINE_KEY_LEN + 1);
         if (p == NULL) {
             return NGX_CONF_ERROR;
         }
 
         llcf->header_filter_src_key = p;
 
+        p = ngx_copy(p, "header_filter_by_lua",
```

Comment:
We should generate a chunk name here and use that. It seems to be another issue that we failed to use the new `ngx_http_lua_gen_chunk_name()` mechanism for these directives.

Reply:
@agentzh 
Yes, the code in `header_filter_by_lua` uses a hard code `=header_filter_by_lua` instead of generated chunk name.
So with `=balancer_by_lua` and some other phases.
Should we change the hard code chunk names and use the `ngx_http_lua_gen_chunk_name()` here too?

Reply:
@spacewander Yes, please.

Reply:
All `xxx_by_lua:...:` need to be rewritten to `xxx_by_lua(nginx.conf:...):...`.

Reply:
@spacewander Yes, definitely. I think we can first merge this PR and address the chunk name thing in a later separate PR.